test/unit/indexer/compare_enumerator_test.rb

```diff
@@ -0,0 +1,133 @@
+require "test_helper"
+require 'indexer/compare_enumerator'
+
+class CompareEnumeratorTest < MiniTest::Unit::TestCase
+  def test_when_matching_keys_exist
+    data_left = { '_root_id' => 'abc', '_root_type' => 'stuff', 'custom' => 'data_left' }
+    data_right = { '_root_id' => 'abc', '_root_type' => 'stuff', 'custom' => 'data_right' }
+
+    stub_scroll_enumerator(left_request: [data_left], right_request: [data_right])
+
+    results = Indexer::CompareEnumerator.new('index_a', 'index_b').map { |a| a }
+    assert_equal results, [[data_left, data_right]]
+  end
+
+  def test_when_key_only_exists_in_left_index
+    data = { '_root_id' => 'abc', '_root_type' => 'stuff', 'custom' => 'data_left' }
+
+    stub_scroll_enumerator(left_request: [data], right_request: [])
+
+    results = Indexer::CompareEnumerator.new('index_a', 'index_b').map { |a| a }
+    assert_equal results, [[data, Indexer::CompareEnumerator::NO_VALUE]]
+  end
+
+
+  def test_when_key_only_exists_in_right_index
+    data = { '_root_id' => 'abc', '_root_type' => 'stuff', 'custom' => 'data_right' }
+
+    stub_scroll_enumerator(left_request: [], right_request: [data])
+
+    results = Indexer::CompareEnumerator.new('index_a', 'index_b').map { |a| a }
+    assert_equal results, [[Indexer::CompareEnumerator::NO_VALUE, data]]
+  end
+
+  def test_with_matching_ids_but_different_types
+    data_left = { '_root_id' => 'abc', '_root_type' => 'stuff', 'custom' => 'data_left' }
+    data_right = { '_root_id' => 'abc', '_root_type' => 'other_stuff', 'custom' => 'data_right' }
+
+    stub_scroll_enumerator(left_request: [data_left], right_request: [data_right])
+
+    results = Indexer::CompareEnumerator.new('index_a', 'index_b').map { |a| a }
+    assert_equal results, [
+      [Indexer::CompareEnumerator::NO_VALUE, data_right],
```

Comment:
Just to check: is this the most useful way of representing this? I'd have thought it would be easier to interpret the results if they were grouped by ID and we checked the type directly, rather than including them in the added/removed count, since adding and removing documents is expected but changing types is not.